doc/dev-guide/src/release-process.md

```diff
@@ -1,66 +1,217 @@
-        > - This release is a hotfix because the last one had a defect:
-        >   increment the *patch* number instead.
-      - Update `Cargo.toml` and `download/Cargo.toml` to have that same new
-        version number, then run `cargo build` and review `Cargo.lock` changes.
+1. If the version strings in `Cargo.toml`s haven't been updated:
+    - Decide what the new version number `$VER_NUM` should be.
+      > **Note:** We always increment the *minor* number unless:
+      > - A major incompatibility has been introduced in this release:
+          >   increment the *major* number instead.
+      > - This release is a hotfix because the last one had a defect:
+          >   increment the *patch* number instead.
+    - Update `Cargo.toml` and `download/Cargo.toml` to have that same new
+      version number, then run `cargo build` and review `Cargo.lock` changes.
       If all looks well, make a commit.
-   2. Update `CHANGELOG.md` accordingly if necessary.
-2. [b/r] After merging the PR made in step 1, in a separate PR:
-   1. Update the commit shasum in `rustup-init.sh` to match the latest commit
-      on `master`.
-3. [b/r] After merging the PR made in step 2, sync `master` to `stable` using
-   `--ff-only`:
-   - `git fetch origin master:master`
-   - `git checkout stable && git merge --ff-only master`
-   - `git push origin HEAD:stable`
-4. [b/r] While you wait for green CI on `stable`, double-check the
-   functionality of `rustup-init.sh` and `rustup-init` just in case.
-5. [b/r] Ensure all of CI is green on the `stable` branch.
-   Once it is, check through a representative proportion of the builds looking
-   for the reported version statements to ensure that
-   we definitely built something cleanly which reports as the right version
-   number when run `--version`.
-6. [r] Make a new PR to the [Rust Blog] adding a new release announcement post.
-7. [b/r] Ping someone in the release team to perform the actual release.
-   They can find instructions in `ci/sync-dist.py`.
-   > **Note:** Some manual testing occurs here, so hopefully they'll catch
-     anything egregious in which case abort the change and roll back.
-8. [b] Once the beta release has happened, post a new topic named "Seeking beta
-   testers for Rustup $VER_NUM" on the [Internals Forum].
-9. [r] Once the official release has happened, prepare and push a tag on the
-   latest `stable` commit.
-   - `git tag -as $VER_NUM -m $VER_NUM` (optionally without `-s` if not GPG
-     signing the tag)
-   - `git push origin $VER_NUM`
-
-[Rust Blog]: https://github.com/rust-lang/blog.rust-lang.org
-[Internals Forum]: https://internals.rust-lang.org
+2. Update `CHANGELOG.md` accordingly if necessary.
+
+Submit the changes in a PR and merge it.
+
+### 3. Update Checksum
+
+After merging the PR in the previous step, update the commit SHA checksum in
+`rustup-init.sh` to match the latest commit on `master`. Submit the change in a
+PR and merge it.
```

Comment:
This step sounds surprising to me. If someone updates this to some other commit, how will we notice? Is there an opportunity to adjust the script as part of CI publishing it?

Reply:
Yeah, for sure. Mostly just flagging it since it sounded off.